### ISSUE TITLE
Unable to create a working build inside a container

### ISSUE THREAD

**Issue by vyasr** [rapidsai/kvikio#33]
I am running inside a Docker container, specifically one created using [rapids-compose](https://github.com/trxcllnt/rapids-compose). I am able to successfully build both the Python and C++ kvikio libraries, but both of them fail at runtime with a cuFile internal error. For example, using the [exact C++ building instructions here](https://github.com/rapidsai/kvikio#c-build-from-source) appears to succeed, but then running the `basic_io` executable gives 
```
(rapids) rapids@compose:~/kvikio/cpp/build$ ./examples/basic_io 
KvikIO config: 
  Compatibility mode: disabled
DriverProperties: 
  Version: 0.0
  Allow compatibility mode: true
  Pool mode - enabled: false, threshold: 4 kb
  Max pinned memory: 33554432 kb
terminate called after throwing an instance of 'kvikio::CUfileException'
  what():  cuFile error at: /home/vyasr/local/rapids/kvikio/cpp/examples/../include/kvikio/file_handle.hpp:117: internal error
Aborted (core dumped)
```
I get the same `cuFile... internal error` if I try to run `pytest` on anything in the `python/tests` directory.

**Comment by madsbk** [comment]
I think this is because cuFile needs `--volume /run/udev:/run/udev:ro` when launching the Docker image. 

@vyasr could you try #34 and see if it fixes the issue by automatically switching to compatibility mode?

**Comment by vyasr** [comment]
@madsbk I confirmed that #34 fixes the issue. I tried a number of changes to my docker launch command yesterday including adding `--volume /run/udev:/run/udev:ro` and I was not able to get cuFile working. I used https://github.com/NVIDIA/MagnumIO/tree/main/gds/docker as a guide and tried various combinations of the docker options suggested there, but I didn't have any luck. However, getting cuFile working inside docker is probably out of scope for this issue, this is probably an issue on my local machine and not in scope for kvikio to fix. Thanks for providing the patch!
